zanshinsdk/test_client.py

```diff
@@ -2518,3 +2518,23 @@ def test_onboard_scan_target_aws_boto3_session(self, request, mock_is_file):
         cf_stacks = cloudformation.describe_stacks(StackName=zanshin_cloudformation_stack_name)
         for cf_stack in cf_stacks['Stacks']:
             cloudformation.delete_stack(StackName=cf_stack['StackName'])
+
+class TestScanTargetSchedule(unittest.TestCase):
+    def test_from_value(self):
+        """
+        Tests the initialization of a new scan target schedule instance using the new enum class, its string equivalent
+        and the old cron-style strings.
+        """
+        for k,v in zanshinsdk.ScanTargetSchedule.__members__.items():
+            self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value(v.value), v)
+            self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value(v), v)
+
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 * * * *'), zanshinsdk.ScanTargetSchedule.ONE_HOUR)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 */6 * * *'), zanshinsdk.ScanTargetSchedule.SIX_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 */12 * * *'), zanshinsdk.ScanTargetSchedule.TWELVE_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 0 * * *'), zanshinsdk.ScanTargetSchedule.TWENTY_FOUR_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 0 * * 0'), zanshinsdk.ScanTargetSchedule.SEVEN_DAYS)
+
+        self.assertRaises(TypeError, zanshinsdk.ScanTargetSchedule.from_value, 1)
+        self.assertRaises(TypeError, zanshinsdk.ScanTargetSchedule.from_value, 1.0)
+        self.assertRaises(ValueError, zanshinsdk.ScanTargetSchedule.from_value, "foo")
```

Comment:
How about checking that '0 0 */8 * *' is not acceptable, for example?

Reply:
Sure, I'll add it for good measure.